src/transforms/projectionpursuit.jl

```diff
@@ -0,0 +1,166 @@
+end
+
+function pindex(transform::ProjectionPursuit, matrix, α)
+  R = rscore(matrix, α)
+  if transform.deg == 1
+      return (3/2) * mean(R)^2
+  else
+    I = (3/2) * mean(R)^2
+    Pⱼ₋₂, Pⱼ₋₁ = ones(size(R)), R
+    for j = 2:transform.deg
+      Pⱼ₋₂, Pⱼ₋₁ = 
+        Pⱼ₋₁, (1/j) * ((2j-1) * R .* Pⱼ₋₁ - (j-1) * Pⱼ₋₂)
+      I += ((2j+1)/2) * (mean(Pⱼ₋₁))^2
+    end
+  end
+  return I
+end
+
+function optimaxis(transform::ProjectionPursuit,matrix)
+  _, q = size(matrix)
+  Iₘₐₓ, iₘₐₓ = 0, 0
+  for i = 1:q
+    eᵢ = [1. * (j==i) for j=1:q]
+    I = pindex(transform, matrix, eᵢ)
+    if I > Iₘₐₓ
+      Iₘₐₓ, iₘₐₓ = I, i
+    end
+  end
+  return [1. *(iₘₐₓ==i) for i = 1:q]
+end
+
+function optimdiag(transform::ProjectionPursuit,matrix, ᾱ)
+  l = length(ᾱ)
+  I₀ = pindex(transform, matrix, ᾱ)
+  for i in 1:l
+    eᵢ = [1. * (j==i) for j=1:l]
+    f₊ = pindex(transform, matrix, (1/√(2+2ᾱ[i]))*(ᾱ + eᵢ))
+    f₋ = pindex(transform, matrix, (1/√(2-2ᾱ[i]))*(ᾱ - eᵢ)) * 
+      (ᾱ[i] != 1)
+    f = (f₊>f₋)*f₊ + (f₊<=f₋)*f₋
+    s = (f₊>f₋)*1 + (f₊<=f₋)*(-1)
+    if f > pindex(transform, matrix, ᾱ)
+      ᾱ = (1/√(2+2s*ᾱ[i]))*(ᾱ + s*eᵢ)
+    end
+  end
+  if pindex(transform, matrix,ᾱ) >= I₀
+    return ᾱ
+  else
+    return optimaxis(transform, matrix)
+  end
+end
+
+function optimgen(transform::ProjectionPursuit, matrix, α₀)
+  f(α) = -pindex(transform, matrix, α)
+  op = optimize(f,α₀)
+  Optim.minimizer(op)
+end
+
+function optim(transform::ProjectionPursuit, matrix)
+  α_axis = optimaxis(transform, matrix)
+  α_diag = optimdiag(transform, matrix, α_axis)
+  optimgen(transform, matrix, α_diag)  
+end
+
+#Transforms the cdf of a convex combination of r.v. to a standard normal cdf.
+function structure_removal(transform::ProjectionPursuit, matrix, α)
+  l = length(α)
+  U = qr(hcat(α,rand(l,l-1)))
+  if norm(diag(U.R)) < transform.tol
+    return print("Error: the random matrix is singular. Try again")
+  end
+  T = matrix * U.Q
+  table = Tables.table(T)
+  newtable, _ = apply(Quantile(),table)
+  T[:,1] = newtable[:,1]
+  return T*U.Q'
+end
+
+#Gives the array that will be used to stop the loop of structure 
+#removal by computing the projection index on k samples of multinomial 
+#distribution (size of sample kxn) along q random directions.
+function gauss_pmatrix(transform::ProjectionPursuit,N,q)
+  I = rand(transform.n,q)
+  d = MvNormal(zeros(q), Diagonal(ones(q)))
+  Q = qr(rand(q,q)).Q
+  for i = 1:transform.n
+    S = rand(d,N)'
+    for j = 1:q
+      I[i,j] = pindex(transform, S, Q[:,j])
+    end
+  end
+  I, Q
+end
+
+#Computes the projection index of each loop dataset (with possibly 
+#modified structure) along q random directions and compares with the 
+#array given by gaussian_projection_matrix.
+function stop_criteria(transform::ProjectionPursuit,matrix,I,Q)
+  k, q = size(I)
+  switch = true
+  for i = 1:q
+    b = pindex(transform,matrix,Q[:,i])
+    switch *= quantile(I[1:k-1,i],transform.perc) > b
```

Comment:
```suggestion
    b = pindex(transform, matrix, Q[:,i])
    switch *= quantile(I[1:k-1,i], transform.perc) > b
```
Always use whitespace to separate arguments of a function.